lisa/operating_system.py

```diff
@@ -150,15 +220,41 @@ def type_name(cls) -> str:
     def name_pattern(cls) -> Pattern[str]:
         return re.compile(f"^{cls.type_name()}$")
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
+        raise NotImplementedError()
+
+    def _query_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> ExecutableResult:
```

Comment:
How about to change the return value to bool? True means exists, False means not exist.

Reply:
> How about to change the return value to bool? True means exists, False means not exist.

Thing is if we are querying for multiple packages, some might be there, some not. Sending the stdout to caller will help check which are not.
Or may be we can send just the list of packages found if not empty.

Reply:
It's better to query only one package each time. I think it happens mostly. If it needs to support querying multiple packages, it can return a list of bool. 
The reason is, the query command may be different for distros, some may return existing package names, some may return a message more than package name like 'package A doesn't exist'. We can envelop them with package commands.

Reply:
Make sense. Hmm lets make it query only one package at a time for now.